server/grpc_ac.go

```diff
@@ -171,6 +171,12 @@ func (s *grpcServer) maybeInline(ctx context.Context, inline bool, slice *[]byte
 			return nil // Not inlined, nothing to do.
 		}
 
+		if (*digest).SizeBytes <= 0 {
+			// Unexpected corner case?
+			*slice = []byte{}
+			return nil
+		}
+
 		if *digest == nil {
```

Comment:
I'm uncertain what your intention was here @mostynb?

It looks suspicious to first use (*digest).SizeBytes and then a few lines further down check for if *digest == nil.

Reply:
It looks your your PR adds this?

Reply:
If we keep this, I think we should move it below the `if *digest == nil {` block.

Reply:
After thinking more about this I believe the corner case can only occur if the ActionResult have **both** an inlined slice (with non-zero length) and **also** a contradicting explicit digest specifying zero length. Because otherwise there would be an early return at the `len(*slice) == 0` check above. I assume invalid digests are managed other ways, and I believe the check should not be added so I reverted it in latest commit.